codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Headers.java

```diff
@@ -69,203 +83,207 @@ public DefaultHttp2Headers() {
      */
     @SuppressWarnings("unchecked")
     public DefaultHttp2Headers(boolean validate) {
-        super(ByteStringValueConverter.INSTANCE, validate ? HTTP2_NAME_VALIDATOR : NameValidator.NOT_NULL);
+        // Case sensitive compare is used because it is cheaper, and header validation can be used to catch invalid
+        // headers.
+        super(CASE_SENSITIVE_HASHER,
+              CharSequenceValueConverter.INSTANCE,
+              validate ? HTTP2_NAME_VALIDATOR : NameValidator.NOT_NULL);
     }
 
     @Override
-    public Http2Headers add(ByteString name, ByteString value) {
+    public Http2Headers add(CharSequence name, CharSequence value) {
         super.add(name, value);
         return this;
     }
 
     @Override
-    public Http2Headers add(ByteString name, Iterable<? extends ByteString> values) {
+    public Http2Headers add(CharSequence name, Iterable<? extends CharSequence> values) {
         super.add(name, values);
         return this;
     }
 
     @Override
-    public Http2Headers add(ByteString name, ByteString... values) {
+    public Http2Headers add(CharSequence name, CharSequence... values) {
```

Comment:
Could we make the base class automatically return the correct type with generics?  It's unfortunate to have to implement every method just for altering the return type.  Alternatively, we could get rid of this class entirely and just have a factory class:

```java
public final class Http2CodecUtil {
...
    public static HttpHeaders newHeaders() {
        // Create HttpHeaders and configure it the way we want.
    }
...
}
```